### PR TITLE
Fix CI verify stage

### PR DESCRIPTION
**Pull request by Niicck**
All the molecule tests now pass locally for me. Let's see how it works in CI.

I cleaned up all the lingering bugs with our molecule tests, plus some new bugs that only showed up in the last couple of months.

Resolves: https://github.com/dokku/ansible-dokku/issues/154

Here's what I did:
- Upgraded molecule test dependencies in `requirements.txt`
  - Older package versions had some breaking dependency issues around python `requests`.
  - Also, the schema for adding molecule plugins/drivers had changed.
- Updated supported distros 
  - Ubuntu 18.04 is out, 22.04 is in. Also added debian 12.
  - Removing Ubuntu 18.04 fixes an error at the `install dokku packages` task: https://github.com/dokku/sshcommand/issues/119
- Updated `dokku_ports.py`
  - The API for managing ports changed with dokku v0.31.0. I put logic in place to handle cases where people haven't upgraded to a version of dokku above 0.31.0
- Replaced depreacted `apt_key` task
  - https://www.jeffgeerling.com/blog/2022/aptkey-deprecated-debianubuntu-how-fix-ansible
- Updated ansible-lint
- Removed debian10 from our CI pipeline.
  - One of dokku's dependencies was removed from debian10, that's why those builds were failing before: https://github.com/dokku/docker-container-healthchecker/issues/129

### CODE REVIEW

File: library/dokku_ports.py [hunk 1]
```diff
@@ -57,13 +57,25 @@
...
     if error is None:
-        for line in output:
-            match = re.match("Proxy port map:(?P<mapping>.+)", line.strip())
-            if match:
-                mappings = match.group("mapping").strip().split(" ")
+        if use_legacy_command():
+            for line in output:
+                match = re.match("Proxy port map:(?P<mapping>.+)", line.strip())
+                if match:
+                    mappings = match.group("mapping").strip().split(" ")
```
Comment (ltalirz) [inline library/dokku_ports.py:72]: this is not your bug but it seems to me, `mappings` here is never set if no match is found.
could you fix this if you don't mind?

**Review by ltalirz — COMMENTED** [review]
Thanks a lot for this work @niicck, much appreciated!

All looks good to me, let's see whether you can figure out the dokku package install as well.

By the way, it's not quite clear to me why CI doesn't run on the branch on your fork but it should be possible to get it to run there (then you don't need to wait for me to click the "approve" button on the workflows).

File: library/dokku_ports.py (hunk 1 above)
Comment (Niicck) [reply library/dokku_ports.py:72]: You know what, `mappings = []` actually already was initialized, I just didn't see it.

https://github.com/dokku/ansible-dokku/pull/166/files#diff-771a074fd2105a1b2651c2dec4070b48246b5ec753f5824031bf3015d07a7f3cR59

**Comment by Niicck** [comment]
I think you can configure the repo to allow non-maintainers to run workflows: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#allowing-select-actions-and-reusable-workflows-to-run

But I don't think it'll be a blocker. I figured out how to run molecule locally with different distros. So I should be able to identify the rest of the issues without needing to run them in the actual CI pipeline

**Comment by Niicck** [comment]
Alright, let's give it another run. I think all the distros should be working.

**Pull request merged**